src/rimecandidate.cpp

```diff
@@ -187,4 +190,19 @@ void RimeCandidateList::triggerAction(const CandidateWord &candidate, int id) {
         }
     }
 }
+
+#ifndef FCITX_RIME_NO_HIGHLIGHT_CANDIDATE
+int RimeCandidateList::globalCursorIndex() const {
+    return -1; // No API available.
+}
+
+void RimeCandidateList::setGlobalCursorIndex(int index) {
+    auto session = engine_->state(ic_)->session(false);
+    if (!session) {
+        throw std::invalid_argument("Invalid session");
```

Comment:
Session availability somehow it's not owned by fcitx. So if no session, just return silently without exception is fine.